### DISCUSSION TITLE
Custom Constraint ON DELETE CASCADE Support?

### DISCUSSION THREAD

**Discussion by davidjadczyk** [simolus3/drift#1121] [Q&A] [answered]
Does Moor support on DELETE CASCADE Constraint?

Because if I delete an Element from a Table with Foreign References I receive:

`no such table: main.workoutcollections (code 1 SQLITE_ERROR): , while compiling: DELETE FROM workouts WHERE id = ?;`

The WorkoutCollections is implemented as following:
```
class WorkoutCollections extends Table {
  IntColumn get id => integer().autoIncrement()();
  IntColumn get position => integer()();
  IntColumn get workoutId => integer()();
  IntColumn get phase => intEnum<WorkoutPhase>()();

  @override
  List<String> get customConstraints =>
      ['FOREIGN KEY(workout_id) REFERENCES workouts(id) ON DELETE CASCADE'];
```

The error occurs when I am trying to Delete an Workout Entry from Workouts Table with the workout_id.

**Comment by simolus3** [comment, marked as answer]
That's interesting. Moor doesn't alter `customConstraints` in any way, so they should work as you'd expect.

It's interesting that the error is about `workoutcollections` not existing. By default, moor would name the table `workout_collections`. Did you rename the table at some point?

**Reply by davidjadczyk** [reply]
After you mentioned that the table would be named it `workout_collections` I recognized I had a constraint from another table on the `workout_collections` in which I wrote `workoutcollections`. So yes it now works as expected. Thank you for guiding me in the right direction 👍🏽

btw: It now also works with `.customConstraint()` as I planned to do
so that it looks like
 `IntColumn get workoutId => integer()
      .customConstraint('REFERENCES workouts (id) ON DELETE CASCADE')();`

**Reply by simolus3** [reply]
Great that you were able to make it work. BTW, if you use `customConstraint` on a column that overrides all other constraints that would be placed by moor. So in your case you'd be missing out on the `NOT NULL` if you don't manually include it in the `customConstraint`.
